src/signal.rs

```diff
@@ -1105,6 +1105,69 @@ mod tool_button {
     }
 }
 
+pub trait ToggleButtonSignals {
+    fn connect_toggled<F: Fn(&Self) + 'static>(&self, f: F) -> u64;
+}
+
+mod toggle_button {
+    use std::mem::transmute;
+    use glib::signal::connect;
+    use glib::translate::*;
+    use ffi::GtkToggleButton;
+    use super::CallbackGuard;
+    use ToggleButton;
+
+    impl super::ToggleButtonSignals for ToggleButton {
+        fn connect_toggled<F: Fn(&Self) + 'static>(&self, f: F) -> u64 {
+            unsafe {
+                let f: Box<Box<Fn(&Self) + 'static>> = Box::new(Box::new(f));
+                connect(self.to_glib_none().0, "toggled",
+                    transmute(void_trampoline), Box::into_raw(f) as *mut _)
+            }
+        }
+    }
+
+    unsafe extern "C" fn void_trampoline(this: *mut GtkToggleButton, f: &Box<Fn(&ToggleButton) + 'static>) {
+        callback_guard!();
+        f(&from_glib_none(this));
+    }
+}
+
+pub trait CellRendererToggleSignals {
+    fn connect_toggled<F: Fn(&Self, &mut TreePath) + 'static>(&self, f: F) -> u64;
+}
+
+mod cell_renderer_toggle {
+    use std::mem::transmute;
+    use std::str;
+    use std::ffi::CStr;
+    use glib::signal::connect;
+    use glib::translate::*;
+    use libc::c_char;
+    use ffi::GtkCellRendererToggle;
+    use super::CallbackGuard;
+    use {CellRendererToggle, TreePath};
+
+    impl super::CellRendererToggleSignals for CellRendererToggle {
+        fn connect_toggled<F: Fn(&Self, &mut TreePath) + 'static>(&self, f: F) -> u64 {
+            unsafe {
+                let f: Box<Box<Fn(&Self, &mut TreePath)>> = Box::new(Box::new(f));
+                connect(self.to_glib_none().0, "toggled",
+                    transmute(string_path_trampoline), Box::into_raw(f) as *mut _)
+            }
+        }
+    }
+
+    unsafe extern "C" fn string_path_trampoline(this: *mut GtkCellRendererToggle, c_str_path: *const c_char,
+            f: &Box<Fn(&CellRendererToggle, &mut TreePath)>) {
+        callback_guard!();
+        let buf = CStr::from_ptr(c_str_path).to_bytes();
+        let string = str::from_utf8(buf).unwrap();
+        let mut path = TreePath::new_from_string(string);
```

Comment:
It seems more straightforward to avoid converting the string back and forth by calling `gtk_tree_path_new_from_string` directly, e.g.
```rust
let path = from_glib_full(ffi::gtk_tree_path_new_from_string(c_str_path));
```